pkg/debuginfo/store.go

```diff
@@ -382,7 +389,16 @@ func validateInput(id string) error {
 }
 
 func (s *Store) find(ctx context.Context, key string) (bool, error) {
-	found := false
+	s.existsCacheMtx.Lock()
+	_, found := s.existsCache[key]
+	s.existsCacheMtx.Unlock()
+
+	if found {
+		// We can't cache the negative result, because the file might be
+		// uploaded in the meantime.
+		return true, nil
+	}
+
 	err := s.bucket.Iter(ctx, key, func(_ string) error {
```

Comment:
Perhaps I'm missing something but why don't we use `get` (not sure what the name is) instead?

Reply:
What would we get? The `metadata.json` or the debuginfo file? This essentially just checks that _anything_ is there.

Reply:
I would suggest the debuginfo file, as it is what we care about the most. We handle a missing metadata file, but a missing debuginfo would be pretty bad

Reply:
From our cloud provider:
- class A: storage.objects.list
- class B: storage.*.get

Class B seems like it's 10x cheaper 😄 
https://cloud.google.com/storage/pricing

Reply:
TIL :)